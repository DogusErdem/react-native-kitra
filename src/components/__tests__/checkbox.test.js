/* eslint-disable no-undef */

import { render, fireEvent } from '@testing-library/react-native';
import { CheckBox } from '../../index';

describe('CheckBox', () => {
  describe('renders with different props', () => {
    it('renders unchecked', () => {
      const tree = render(<CheckBox testID="checkbox" value={false} />).toJSON();

      expect(tree).toMatchSnapshot();
    });

    it('renders checked', () => {
      const tree = render(<CheckBox testID="checkbox" value />).toJSON();

      expect(tree).toMatchSnapshot();
    });

    it('renders disabled', () => {
      const tree = render(<CheckBox testID="checkbox" disabled />).toJSON();

      expect(tree).toMatchSnapshot();
    });
    it('renders custom icon color', () => {
      const tree = render(<CheckBox testID="checkbox" iconColor="red" value />).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe('calls the correct function when pressed', () => {
    it('calls onPress with the opposite of the current status when pressed', () => {
      const onPress = jest.fn();
      const { getByTestId } = render(<CheckBox testID="checkbox" onPress={onPress} value />);
      const checkbox = getByTestId('checkbox');
      fireEvent.press(checkbox);

      expect(onPress).toHaveBeenCalledWith(false);
    });

    it('calls onChange when the checkbox is pressed', () => {
      const onChange = jest.fn();
      const { getByTestId } = render(<CheckBox testID="checkbox" onChange={onChange} value={false} />);

      fireEvent.press(getByTestId('checkbox'));

      expect(onChange).toHaveBeenNthCalledWith(1, false);
      expect(onChange).toHaveBeenNthCalledWith(2, true);
    });
  });
});
