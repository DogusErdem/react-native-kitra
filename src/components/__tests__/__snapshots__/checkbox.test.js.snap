// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox renders with different props renders checked 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": undefined,
      "borderColor": undefined,
      "borderRadius": 4,
      "borderWidth": 1,
      "height": 20,
      "justifyContent": "center",
      "opacity": 1,
      "width": 20,
    }
  }
  testID="checkbox"
>
  <View
    entering={BaseAnimationMock {}}
    exiting={BaseAnimationMock {}}
  >
    <Text
      accessibilityElementsHidden={true}
      allowFontScaling={false}
      importantForAccessibility="no-hide-descendants"
      pointerEvents="none"
      selectable={false}
      style={
        [
          {
            "color": undefined,
            "fontSize": 12,
          },
          [
            {
              "lineHeight": 12,
            },
            {
              "backgroundColor": "transparent",
            },
            undefined,
          ],
          {
            "fontFamily": "Octicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`CheckBox renders with different props renders custom icon color 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": undefined,
      "borderColor": undefined,
      "borderRadius": 4,
      "borderWidth": 1,
      "height": 20,
      "justifyContent": "center",
      "opacity": 1,
      "width": 20,
    }
  }
  testID="checkbox"
>
  <View
    entering={BaseAnimationMock {}}
    exiting={BaseAnimationMock {}}
  >
    <Text
      accessibilityElementsHidden={true}
      allowFontScaling={false}
      importantForAccessibility="no-hide-descendants"
      pointerEvents="none"
      selectable={false}
      style={
        [
          {
            "color": "red",
            "fontSize": 12,
          },
          [
            {
              "lineHeight": 12,
            },
            {
              "backgroundColor": "transparent",
            },
            undefined,
          ],
          {
            "fontFamily": "Octicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`CheckBox renders with different props renders disabled 1`] = `
<View
  accessibilityState={
    {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": undefined,
      "borderColor": undefined,
      "borderRadius": 4,
      "borderWidth": 1,
      "height": 20,
      "justifyContent": "center",
      "opacity": 1,
      "width": 20,
    }
  }
  testID="checkbox"
/>
`;

exports[`CheckBox renders with different props renders unchecked 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": undefined,
      "borderColor": undefined,
      "borderRadius": 4,
      "borderWidth": 1,
      "height": 20,
      "justifyContent": "center",
      "opacity": 1,
      "width": 20,
    }
  }
  testID="checkbox"
/>
`;
